AddressBook.java

```diff
@@ -0,0 +1,1317 @@
+        return split.length == 2 ? split : new String[] { split[0] , "" }; // else case: no parameters
+    }
+
+    /**
+     * Constructs a generic feedback message for an invalid command from user, with instructions for correct usage.
+     *
+     * @param correctUsageInfo message showing the correct usage
+     * @return invalid command args feedback message
+     */
+    private static String getMessageForInvalidCommandInput(String userCommand, String correctUsageInfo) {
+        return String.format(MESSAGE_INVALID_COMMAND_FORMAT, userCommand, correctUsageInfo);
+    }
+
+    /**
+     * Adds a person (specified by the command args) to the address book.
+     * The entire command arguments string is treated as a string representation of the person to add.
+     *
+     * @param commandArgs full command args string from the user
+     * @return feedback display message for the operation result
+     */
+    private static String executeAddPerson(String commandArgs) {
+        // try decoding a person from the raw args
+        final Optional<String[]> decodeResult = decodePersonFromString(commandArgs); //will return an optional data type with string array parameter. The string indexes are as follows: name, phone and email. 
+
+        // checks if args are valid (decode result will not be present if the person is invalid)
+        if (!decodeResult.isPresent()) {
+            return getMessageForInvalidCommandInput(COMMAND_ADD_WORD, getUsageInfoForAddCommand());
+        }
+
+        // add the person as specified
+        final String[] personToAdd = decodeResult.get();
+        addPersonToAddressBook(personToAdd);
+        HashMap<personProperty, String> personOnAdd = new HashMap<personProperty, String>();
+        personOnAdd.put(personProperty.NAME, personToAdd[0]);
+        personOnAdd.put(personProperty.PHONE, personToAdd[1]);
+        personOnAdd.put(personProperty.EMAIL, personToAdd[2]);
+        return getMessageForSuccessfulAddPerson(personOnAdd);
+    }
+    private static void executeEditRequest(String commandArgs){
+    	String[] field= commandArgs.trim().split("\\s+", 3);
+    	
+    	char check_digit = field[0].charAt(0);
+    	int index;
+    	String key;
+    	String newValue=field[2];
+    	if(Character.isDigit(check_digit)){
+    		index=Integer.parseInt(field[0]);
+    		key=field[1].toUpperCase();
+    	}
+    	else {
+    		index= Integer.parseInt(field[1]);
+    		key=field[0].toUpperCase();
+    	}
+    	
+    	index-=1;
+    	
+    	switch(key){
+    	case "NAME":{
+    		ALL_PERSONS.get(index).put(personProperty.NAME, newValue);
+    	}
+    	break;
+    	case "EMAIL":{
+    		ALL_PERSONS.get(index).put(personProperty.EMAIL, newValue);
+    	}
+    	break;
+    	case "PHONE":{
+    		ALL_PERSONS.get(index).put(personProperty.PHONE, newValue);
+    	}
+    	break;
+    	}
+    	
```

Comment:
would be better if syntax for edit is similar to add since users will have fewer command syntax to remember. also, having same syntax would mean you can use the same parsing method for both add and edit commands. current code is kinda hard to understand- consider improving SLAP/not using magic numbers